ifsbench/env.py

```diff
@@ -171,35 +166,19 @@ class DefaultEnvPipeline(EnvPipeline):
         If overrides contains entries that are not EnvOverride objects.
     """
 
-    def __init__(
-        self,
-        handlers: Optional[List[EnvHandler]] = None,
-        env_initial: Optional[Dict[str, str]] = None,
-    ):
-        if handlers:
-            self._handlers = list(handlers)
-        else:
-            self._handlers = []
-
-        for handler in self._handlers:
-            if not isinstance(handler, EnvHandler):
-                raise ValueError("Only EnvHandler objects are accepted!")
-
-        if env_initial is not None:
-            self._env_initial = dict(env_initial)
-        else:
-            self._env_initial = {}
+    handlers: List[EnvHandler] = Field(default_factory=list)
+    env_initial: Optional[Dict[str, Optional[str]]] = Field(default_factory=dict)
 
     def add(self, handler: Union[EnvHandler, List[EnvHandler]]) -> None:
         if isinstance(handler, EnvHandler):
-            self._handlers.append(handler)
+            self.handlers.append(handler)
         else:
-            self._handlers += handler
+            self.handlers += handler
```

Comment:
[optional] This line is currently not covered by tests - which should be the scenario that `handler` is a list. It might be easy to capture this by adding a test parametrization which simply uses a 1-entry list. But that lack of coverage was there before, so not making this a requirement here